tests_e2e/test_suites/images.yml

```diff
@@ -0,0 +1,70 @@
+#
+# Image sets are used to group images
+#
+image-sets:
+   # Endorsed distros that are tested on the daily runs
+   endorsed:
+#
+# TODO: Add CentOS 6.10 and Debian 8
+#
+#      - "centos_610"
+      - "centos_79"
+#      - "debian_8"
+      - "debian_10"
+      - "debian_9"
+      - "suse_12"
+      - "mariner_1"
+      - "mariner_2"
+      - "mariner_2_arm64"
+      - "suse_15"
+      - "rhel_78"
+      - "rhel_82"
+      - "ubuntu_1604"
+      - "ubuntu_1804"
+      - "ubuntu_2004"
+
+#
+# An image can be specified by a string giving its urn, as in
+#
+#    ubuntu_2004: "Canonical 0001-com-ubuntu-server-focal 20_04-lts latest"
+#
+# or by an object with 3 properties: urn, locations and vm_sizes, as in
+#
+#   mariner_2_arm64:
+#      urn: "microsoftcblmariner cbl-mariner cbl-mariner-2-arm64 latest"
+#      locations:
+#         - "eastus"
+#      vm_sizes:
+#         - "Standard_D2pls_v5"
+#
+# 'urn' is required, while 'locations' and 'vm_sizes' are optional. The latter
+# two properties can be used to specify that the image is available only in
+# some locations, or that it can be used only on some VM sizes.
+#
+# URNs follow the format '<Publisher> <Offer> <Sku> <Version>' or
+# '<Publisher>:<Offer>:<Sku>:<Version>'
+#
+images:
+#
+# TODO: Add CentOS 6.10 and Debian 8
+#
+#   centos_610: "OpenLogic CentOS 6.10 latest"
+   centos_79: "OpenLogic CentOS 7_9 latest"
+#   debian_8: "credativ Debian 8 latest"
+   debian_9: "credativ Debian 9 latest"
+   debian_10: "Debian debian-10 10 latest"
+   mariner_1: "microsoftcblmariner cbl-mariner cbl-mariner-1 latest"
+   mariner_2: "microsoftcblmariner cbl-mariner cbl-mariner-2 latest"
+   mariner_2_arm64:
+      urn: "microsoftcblmariner cbl-mariner cbl-mariner-2-arm64 latest"
+      locations:
```

Comment:
why do we need list here? I don't see a use case? and also, your implementation is only reading first value? same goes for vm_sizes?

Reply:
An image may be available only in a few regions. 

Say an image I is available only on regions R1 and R2. Say we have test suites T1, which needs to run on R1, T2, which needs to run on R2, and T3, which can run on any region.

In that case, we would use R1 for T1, R2 for T2, and for T3 we don't care about the region, so we just use whatever value is item 0 of the list.

Reply:
That is not implemented right where you are not taking region based on test_suite. The example you provided could be achieved by setting region on test_suite.yml itself. I still don't see multiple in images list.

Reply:
it's done by the combinator:
```

            for image in images_info:
                # If the suite specifies a location, use it. Else, if the image specifies a list of locations, use
                # any of them. Otherwise, use the default location.
                if suite_info.location != '':
                    location = suite_info.location
                elif len(image.locations) > 0:
                    location = image.locations[0]
                else:
                    location = AgentTestSuitesCombinator._DEFAULT_LOCATION
```

Reply:
> The example you provided could be achieved by setting region on test_suite.yml itself.

No, setting the region on the test suite has a different meaning. That means the suite must run on that location. When set on the image it means the image is available only on a subset of regions. They are different (and could have different values)

Reply:
My concern is this, every suite gets same location. What is the point of having multiple.

```
                elif len(image.locations) > 0:
                    location = image.locations[0]
```

Reply:
why is that a concern? for sure we wouldn't be overloading that region with out tests :)

Reply:
I mean you said we will have leverage to pick the location based on test_suite but implementation is not reflecting that.

Reply:
Anyway, I'm ok for now. we will see when we have that use case.

Reply:
it's done by the combinator. see the code snippet i posted above and run it over the example i mentioned 

> Say an image I is available only on regions R1 and R2. Say we have test suites T1, which needs to run on R1, T2, which needs to run on R2, and T3, which can run on any region.

>In that case, we would use R1 for T1, R2 for T2, and for T3 we don't care about the region, so we just use whatever value is item 0 of the list.

Reply:
According to the schema I could have R3, R4, R5 in the image location. Lets say I didn't define any region on T3. So T3 would pick R3 and say new T4 also don't have region defined. It will pick R3 only, where is the case any test case would pick R4 and R5.

Reply:
None. 

That the image is available on R3, R4, and R5 doesn't mean we need to use all of them. If T4 doesn't specify a location then using item 0 (R3) is as good as using item 1 (R4), or item 2 (R5).

If later on, another suite T9 needs to run on R5, then R5 will be used.

Reply:
Maybe I didn't understand completely. I still don't see list is useful here. I'm approving it and we can discuss when you provide another session on this pipeline.

Reply:
The list is needed if there are test suites that need to run on different locations.

If T1 needs to run on R1 and T2 needs to run on R2, and the image is available only on R1 and R2, if we don't use a list then you need to chose between R1 and R2, and then you can run only 1 test: either T1 or T2.

Yes, we can go over this in the presentation.